### DISCUSSION TITLE
Make a Line the Focused element

### DISCUSSION THREAD

**Discussion by matthewcrews** [fsprojects/Avalonia.FuncUI#332] [Q&A] [answered]
I am working on a problem where I need to create a `onKeyPressed` event for a line. For this to work I've been told that I need to make the element the focus for it to respond to events. I see how to make something "Focusable" but not how to focus on the element. Here is some example code:

```fsharp
Line.create [
    Line.startPoint (location * state.Zoom + state.WindowPosition)
    Line.endPoint state.PointerPosition
    Line.focusable true // Here I make the Line something I can focus on
    Line.stroke Dims.Line.color
    Line.strokeThickness 2.0
    Line.onKeyDown (fun e ->
        e.Handled <- true
        if e.Key = Key.Escape then
            Msg.Escape |> dispatch)
]
```

I also tried to create the line and then set a `Focused` property but I don't see one.

```fsharp
let line = Line.create [
    Line.startPoint (location * state.Zoom + state.WindowPosition)
    Line.endPoint state.PointerPosition
    Line.focusable true
    Line.stroke Dims.Line.color
    Line.strokeThickness 2.0
    Line.onKeyDown (fun e ->
        e.Handled <- true
        if e.Key = Key.Escape then
            Msg.Escape |> dispatch)
]
line.Focused <- true // This doesn't exist but it was my first thought
line
```

I'm also open to adding the `onKeyDown` handler to the Canvas element that I'm drawing the line one. That raises the same problem though. I asked a question on the [AvaloniaUI GitHub](https://github.com/AvaloniaUI/Avalonia/discussions/12163#discussioncomment-6431369) which is what lead me to asking about setting the focus.

**Comment by JaggerJo** [comment, marked as answer]
IIRC not all controls can be in focus, they need to be `IInputElements`.

I would build my own way of knowing which line/box is selected.
